### PR TITLE
Tilemaps

### PR DESCRIPTION
**Pull request by MaximGorbach**
Added tilemap support with capability for multiple tilemap and object layers, as well as using multiple tilesets.
Tilemaps must be exported from 'Tiled' in `.json` format, **with all tilesets embedded**. For more detailed documentation see the docs on `engine.TileMapJSONParser.parse`.
You can add new tilemap layers by going through the 'add layer' button on the UI.

> **Note:** Actually importing a tilemap is a bit scuffed at the moment, as the files have to be loaded from the `assets` directory currently to work. This means that you have to manually put the tileset images into the assets directory and change the paths in the `json` to point to them (if they don't already). This will require changing once we sort out how textures are loaded. 

> Another issue is that the tilemap `.json` can't be moved as this will mess up the relative paths to the tilesets contained within. To move it, you must re-export from Tiled into the desired location.

### CODE REVIEW

**Review by Louis-Tarvin — CHANGES_REQUESTED** [review]
Looks good, however the add layer dialogue seems bugged as there are duplicate fields.

File: pixelot/src/app/left-sidebar/left-sidebar.component.ts [hunk 1]
```diff
@@ -227,17 +227,39 @@ export class LeftSidebarComponent {
...
+      console.log(result);
+
+      let new_layer: engine.RenderLayer;
+
+      if (type == 'tilemap') {
+        if (result.path == '') {
+          this._snackBar.open('Please enter a path', 'Close', {
+            duration: 5000
+          });
+          return;
```
Comment (Louis-Tarvin) [inline pixelot/src/app/left-sidebar/left-sidebar.component.ts:252]: I get console errors if I don't include a path or include an invalid json file. It looks like you've added the error handling in place here but for some reason it's not triggering?

File: pixelot/src/app/left-sidebar/left-sidebar.component.ts [hunk 2]
```diff
@@ -227,17 +227,39 @@ export class LeftSidebarComponent {
...
+        }
 
-      engine.Renderer.addLayer(new engine.SpriteLayer(), result, engine.SceneManager.currentScene);
-      this.sceneData.addLayer(this.scene.name, result);
+        new_layer = engine.TileMapJSONParser.parse(result.path);
```
Comment (Louis-Tarvin) [inline pixelot/src/app/left-sidebar/left-sidebar.component.ts:255]: Maybe wrap this in a try-catch and display a snackbar like above if parsing fails

**Review by Louis-Tarvin — CHANGES_REQUESTED** [review]
Just a couple of minor error handling suggestions. Other than that seems good

**Comment by MaximGorbach** [comment]
I've added the suggested error handling, and also fixed a bug where right clicking an entity would open the layer menu lol.

**Review by Louis-Tarvin — APPROVED** [review]

**Pull request merged**